### PR TITLE
HaikuPorter/ShellScriptlet.py: Fixed infinite recursion

### LINKED ISSUE #293: Bootstrap builds broken since commit 35a32ad20a9f396625c7e176f7bac155f3e71378
It seems this commit broke bootstrap builds. Didn't investigate that much further but when trying to build a ppc or arm64 bootsteap image it gets to building zlib_bootstrap package and when building it it seems that the wrapper script gets into an infinite recursion when trying to find cmake until it is killed / segfaults. After reverting this commit it works again.

I think the problem is that the function is called cmake and command -v will return that function name instead of searching the cmake binary. Below is a small test case sctipt, try it with bash -x and you'll see that it recurses infinitely, but if you comment out the command -v line and uncomment the which line it finds the cmake and excutes it.

```
#!/bin/bash

cmake()
{
	echo "start"
	CMAKE="/temp/cmake"
	if [ ! -f $CMAKE ]; then
		CMAKE=$(command -v cmake)
		#CMAKE=$(which cmake)
		echo "command -v returns \"$CMAKE\""
	fi
	echo "CMAKE = $CMAKE"
	$CMAKE 
	echo "end"
}

cmake

```

### PR DESCRIPTION
**Pull request by ilzu**
…on bootstrap builds.

Fixes #293

### CODE REVIEW

File: HaikuPorter/ShellScriptlets.py [hunk 1]
```diff
@@ -331,7 +331,8 @@ def getShellVariableSetters(shellVariables):
 
 	CMAKE=$portPackageLinksDir/cmd~cmake/bin/cmake
 	if [ ! -f $CMAKE ]; then
-		CMAKE=$(command -v cmake)
+		CMAKE=$(type -Pp cmake)
```
Comment (jmairboeck) [inline HaikuPorter/ShellScriptlets.py:334]:
```suggestion
		CMAKE=$(type -P cmake)
```

File: HaikuPorter/ShellScriptlets.py [hunk 2]
```diff
@@ -348,7 +349,7 @@ def getShellVariableSetters(shellVariables):
 
 	MESON=$portPackageLinksDir/cmd~meson/bin/meson
 	if [ ! -f $MESON ]; then
-		MESON=$(command -v meson)
+		MESON=$(type -Pp meson)
```
Comment (jmairboeck) [inline HaikuPorter/ShellScriptlets.py:352]:
```suggestion
		MESON=$(type -P meson)
```

File: HaikuPorter/ShellScriptlets.py [hunk 3]
```diff
@@ -331,7 +331,8 @@ def getShellVariableSetters(shellVariables):
 
 	CMAKE=$portPackageLinksDir/cmd~cmake/bin/cmake
 	if [ ! -f $CMAKE ]; then
-		CMAKE=$(command -v cmake)
+		CMAKE=$(type -Pp cmake)
+        echo "CMAKE = $CMAKE"
```
Comment (jmairboeck) [inline HaikuPorter/ShellScriptlets.py:335]: Maybe remove this debug output?

**Review by jmairboeck — APPROVED** [review]

File: HaikuPorter/ShellScriptlets.py (hunk 3 above)
Comment (ilzu) [reply HaikuPorter/ShellScriptlets.py:335]: Damn, I forgot that there.

File: HaikuPorter/ShellScriptlets.py (hunk 1 above)
Comment (ilzu) [reply HaikuPorter/ShellScriptlets.py:334]: I think it needs both -P (search only in path, not in functions and aliases) and -p (just return the file name, or nothing)

Comment (ilzu) [reply HaikuPorter/ShellScriptlets.py:334]: perhaps even -fPp (f = suppress shell function lookup)

Comment (jmairboeck) [reply HaikuPorter/ShellScriptlets.py:334]: It seems to work fine `-Pp`. The extra `p` is no big deal. Thanks for fix!

**Pull request merged**